src/features/textEditor/components/tipTapNodes/notionBlock/plugins/blockBrowsingPlugin.ts

```diff
@@ -232,8 +232,24 @@ function moveBlocksDown({ depthDifference, endPos, selection, tr, dispatch }: Bl
   // If the last selected node has a sibling
   if (indexInParent < parent.childCount - 1) {
     const sibling = parent.child(indexInParent + 1);
-    // If the sibling is collapsed, move selection after the sibling
-    if (sibling.attrs.type === 'collapsible' && sibling.attrs.collapsed) {
+    // If the sibling is a list item, an uncollapsed collapsible block or has children, make selected blocks its child
+    if (
+      (LIST_ITEM_TYPES.includes(sibling.attrs.type as string) && !sibling.attrs.collapsed) ||
+      (!LIST_ITEM_TYPES.includes(sibling.attrs.type as string) && sibling.childCount > 1)
+    ) {
+      const sliceToMove = tr.doc.slice(selection.from + depthDifference, endPos);
+
+      tr.delete(selection.from + depthDifference, endPos);
+      const insertPos = tr.mapping.map(endPos + 1 + sibling.firstChild!.nodeSize);
+      tr.replace(insertPos, insertPos, sliceToMove);
+
+      if (selection.from === selection.anchor) {
+        tr.setSelection(new BlockSelection(tr.doc.resolve(insertPos), tr.doc.resolve(insertPos + selectionRange)));
+      } else {
+        tr.setSelection(new BlockSelection(tr.doc.resolve(insertPos + selectionRange), tr.doc.resolve(insertPos)));
+      }
+    } else {
+      // Otherwise, move selection after the sibling
```

Comment:
This might look like a lot of changes but that is because I inverted the if and else blocks: it feels more natural to treat the special case where the block is indented first, and then have the default case (block moving before or after its sibling) in the else block